apluslms_roman/backends/docker.py

```diff
@@ -79,17 +81,38 @@ def prepare(self, task, observer):
         for step in task.steps:
             observer.step_preflight(step)
             image, tag = step.img.split(':', 1)
+
+            day = timedelta(days=1)
+            cache = CacheFile.load(allow_missing=True)
+            last_updated = cache.mlget('images.%s' % (step.img,), None)
+            last_updated = (day if not last_updated
+                else datetime.now() - last_updated)
+
             try:
-                img = client.images.get(step.img)
+                client.images.get(step.img)
+                img_found = True
             except docker.errors.ImageNotFound:
+                img_found = False
+
+            if not img_found or last_updated >= day:
                 observer.step_running(step)
-                observer.manager_msg(step, "Downloading image {}".format(step.img))
+                if img_found:
+                    observer.manager_msg(step,
+                        "Checking for updates for {}".format(step.img))
+                observer.manager_msg(step,
+                    "Downloading image {}".format(step.img))
                 try:
-                    img = client.images.pull(image, tag)
+                    client.images.pull(image, tag)
+                    cache.mlset('images.%s' % (step.img,), datetime.now())
+                    cache.save()
```

Comment:
You could call `save()` one at the end of the function and not in every loop. E.g. add it to try-finally.